File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def bits(self, n: int) -> int:
        sum = 0;
        while n != 0:
            n = n & (n-1)
            sum += 1
        return sum


    def countBits(self, n: int) -> List[int]:
        res = []
        for i in range(0, n+1):
            res.append(self.bits(i))
        return res

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int res = 0;
        int sum = 0;
        Map<Integer, Integer> map = new HashMap<>();
        map.put(0, 1);

        for (int i:nums) {
            sum += i;
            int diff = sum - k;
            res += map.getOrDefault(diff, 0);
            map.put(sum, map.getOrDefault(sum, 0)+1);
        }
        return res;
    }
}